syntax = "proto3";

package auth;
import "google/protobuf/wrappers.proto";
service AuthService {
    rpc CheckAuth (CheckAuthRequest) returns (CheckAuthResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc Logout (LogoutRequest) returns (LogoutResponse);
}

message Credentials {
    string user_id = 1;
}
message Tokens {
    string access_token = 1;
    string refresh_token = 2;
}

message CheckAuthRequest {
    string ip = 1;
    string agent = 2;
    Tokens tokens = 3;
}

message CheckAuthResponse {
    bool is_login = 1;
    optional Tokens new_tokens = 2;
    optional Credentials credentials = 4;
}

message LoginRequest {
    string ip = 1;
    string agent = 2;
    string user_email = 3;
    string password = 4;
}
message LoginResponse {
    bool is_login = 1;
    optional Tokens tokens = 2;
}
message LogoutRequest {
    Credentials credentials = 1;
}
message LogoutResponse {
    bool is_logout = 1;
}